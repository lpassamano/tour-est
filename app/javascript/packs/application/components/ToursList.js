import React, { Component } from "react";
import { connect } from "react-redux";
import * as tourActions from "../redux/tours/actions";
import * as tourSelectors from "../redux/tours/selectors";

export class ToursList extends Component {
  componentDidMount() {
    this.props.listTours();
  }

  render() {
    const { isFetching, tours } = this.props;

    if (isFetching) {
      return <p>loading.... please wait!</p>;
    }

    return <div>{tours.map(tour => <h4 key={tour.id}>{tour.title}</h4>)}</div>;
  }
}

const mapStateToProps = state => ({
  tours: tourSelectors.getTours(state),
  isFetching: tourSelectors.isFetching(state)
});

const mapDispatchToProps = { listTours: tourActions.listTours };

const enhance = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default enhance(ToursList);
