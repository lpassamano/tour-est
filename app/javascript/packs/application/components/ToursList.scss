.cc-name {
  font-style: italic;
  color: gray;
}

h4.tour-title {
  margin-bottom: 0.3rem;
}
