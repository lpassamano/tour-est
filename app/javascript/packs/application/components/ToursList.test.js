import ToursList from "./ToursList";
import React from "react";
import { mount } from "enzyme";

describe("<ToursList />", () => {
  test("render", () => {
    const tours = [
      { id: 1, title: "Tour 1" },
      { id: 2, title: "Tour 2" },
      { id: 3, title: "Tour 3" }
    ];
    const component = mount(<ToursList tours={tours} />);
    expect(component).toMatchSnapshot();
  });
});
