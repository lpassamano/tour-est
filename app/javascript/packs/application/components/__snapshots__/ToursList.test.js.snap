// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToursList /> render 1`] = `
<div>
  <div
    className="media-object-basics-example"
    key="1"
  >
    <div>
      <h4
        key="1"
      >
        <[object Object]
          to="/tours/1"
        >
          Tour 1
        </[object Object]>
      </h4>
    </div>
    <hr />
  </div>
  <div
    className="media-object-basics-example"
    key="2"
  >
    <div>
      <h4
        key="2"
      >
        <[object Object]
          to="/tours/2"
        >
          Tour 2
        </[object Object]>
      </h4>
    </div>
    <hr />
  </div>
  <div
    className="media-object-basics-example"
    key="3"
  >
    <div>
      <h4
        key="3"
      >
        <[object Object]
          to="/tours/3"
        >
          Tour 3
        </[object Object]>
      </h4>
    </div>
    <hr />
  </div>
</div>
`;
